packages/modeling/src/operations/extrusions/extrudeHelical.js

```diff
@@ -0,0 +1,95 @@
+const { TAU } = require('../../maths/constants')
+const slice = require('./slice')
+const mat4 = require('../../maths/mat4')
+const extrudeFromSlices = require('./extrudeFromSlices')
+const geom2 = require('../../geometries/geom2')
+
+/**
+ * Perform a helical extrude of the geometry, using the given options.
+ *
+ * @param {Object} options - options for extrusion
+ * @param {Number} [options.angle=TAU] - angle of the extrusion (RADIANS) positive for right-hand rotation, negative for left-hand
+ * @param {Number} [options.startAngle=0] - start angle of the extrusion (RADIANS)
+ * @param {Number} [options.pitch=10] - elevation gain for each turn
+ * @param {Number} [options.height] - total height of the helix path. Ignored if pitch is set.
+ * @param {Number} [options.endOffset=0] - offset the final radius of the extrusion, allowing for tapered helix, and or spiral
+ * @param {Number} [options.segments=32] - number of segments of the extrusion
+ * @param {geom2} geometry - the geometry to extrude
+ * @returns {geom3} the extruded geometry
+ * @alias module:modeling/extrusions.extrudeHelical
+ *
+ * @example
+ * const myshape = extrudeHelical(
+ *  {
+ *      angle: Math.PI * 4,
+ *      pitch: 10,
+ *      segments: 64
+ *  },
+ *  circle({size: 3, center: [10, 0]})
+ * )
+ */
+const extrudeHelical = (options, geometry) => {
+  const defaults = {
+    angle: TAU,
+    startAngle: 0,
+    pitch: 10,
+    endOffset: 0,
+    segments: 32
+  }
+  const { angle, endOffset, segments, startAngle } = Object.assign({}, defaults, options)
+
+  let pitch
+  // ignore height if pitch is set
+  if(!options.pitch && options.height) {
+    pitch = options.height / (angle / TAU)
+  } else {
+    pitch = options.pitch ? options.pitch : defaults.pitch
+  }
+
+  if (segments < 2) throw new Error('segments must be greater than 1')
```

Comment:
hmm... i don't see valid results coming from 2 segments. maybe minimum is 3.

Reply:
Thinking about this, it needs at least 3 segments for each complete revolution, in order to get a valid geometry. Even if it's a partial revolution, so we round up.

To see this, we can find an angle $\alpha$ for every number of segments $n \in \mathbb{Z}^{+}$ so that

$$
{\alpha \over n} = \pi
\Leftrightarrow
\alpha = \pi n
$$

Which means that for every number of segments we can find an angle that will project all the slices in the same plane, resulting in incorrect (flat) geometry. Requiring that the number of segments is larger than any arbitrary value can never solve this problem.